### PR TITLE
Better support managing app-identity associations.

### PR DESCRIPTION
**Pull request by jortel**
Better support managing app-identity associations.
Identities are _many-2-many_ with applications.  They need to be managed like:

1. create identities.
2. create applications with identity FKs listed.
and/or
2. update application with updated identity FKs listed.

The association needs to be managed from the application.  Essentially, manage the association like we do for Tags.

---

Adds `Ref` for FKs.

### CODE REVIEW

File: api/identity.go [hunk 1]
```diff
@@ -27,7 +27,6 @@ func (h IdentityHandler) AddRoutes(e *gin.Engine) {
 	e.GET(IdentityRoot, h.Get)
 	e.PUT(IdentityRoot, h.Update)
 	e.DELETE(IdentityRoot, h.Delete)
-	e.POST(AppIdentitiesRoot, h.CreateForApplication)
```
Comment (jortel) [inline api/identity.go:30]: This no longer makes sense.

File: api/identity.go [hunk 2]
```diff
@@ -80,35 +79,6 @@ func (h IdentityHandler) List(ctx *gin.Context) {
 	ctx.JSON(http.StatusOK, resources)
 }
 
-// ListByApplication  godoc
```
Comment (jortel) [inline api/identity.go:83]: Just moved to later in the file.
Following a pattern of grouping all _sub-resource_ endpoints after all the _main_ endpoints.

File: api/application.go [hunk 3]
```diff
@@ -168,11 +175,18 @@ func (h ApplicationHandler) Update(ctx *gin.Context) {
 		return
 	}
 	m := r.Model()
+	nID, _ := strconv.Atoi(id)
+	m.ID = uint(nID)
```
Comment (jortel) [inline api/application.go:179]: The PK needs to be set to manage the associations.

File: api/identity.go [hunk 4]
```diff
@@ -136,42 +106,6 @@ func (h IdentityHandler) Create(ctx *gin.Context) {
...
-// @summary Create an identity for an application.
-// @description Create an identity for an application.
-// @tags create
-// @accept json
-// @produce json
-// @success 201 {object} Identity
-// @router /application-inventory/application/{id}/identities [post]
-// @param id path int true "Application ID"
-// @param identity body Identity true "Identity data"
-func (h IdentityHandler) CreateForApplication(ctx *gin.Context) {
```
Comment (jortel) [inline api/identity.go:149]: Associated endpoint removed.

File: api/identity.go [hunk 5]
```diff
@@ -226,19 +160,47 @@ func (h IdentityHandler) Update(ctx *gin.Context) {
 	ctx.Status(http.StatusNoContent)
 }
 
+// ListByApplication  godoc
+// @summary List identities for an application.
```
Comment (jortel) [inline api/identity.go:164]: Just moved from earlier in the file.

File: api/identity.go [hunk 6]
```diff
@@ -226,19 +160,47 @@ func (h IdentityHandler) Update(ctx *gin.Context) {
...
 	Resource
-	Kind          string `json:"kind" binding:"required"`
-	Name          string `json:"name" binding:"required"`
-	Description   string `json:"description"`
-	User          string `json:"user"`
-	Password      string `json:"password"`
-	Key           string `json:"key"`
-	Settings      string `json:"settings"`
-	Encrypted     string `json:"encrypted"`
-	ApplicationID uint   `json:"application"`
```
Comment (jortel) [inline api/identity.go:241]: ApplicationID not correct for _many-to-many_.

File: api/identity.go [hunk 7]
```diff
@@ -253,22 +215,20 @@ func (r *Identity) With(m *model.Identity) {
...
 	m = &model.Identity{
-		Kind:          r.Kind,
-		Name:          r.Name,
-		Description:   r.Description,
-		User:          r.User,
-		Password:      r.Password,
-		Key:           r.Key,
-		Settings:      r.Settings,
-		Encrypted:     r.Encrypted,
-		ApplicationID: r.ApplicationID,
```
Comment (jortel) [inline api/identity.go:271]: Removed: not correct for _many-to-many_.

File: hack/add/all.sh [hunk 8]
```diff
@@ -10,12 +10,12 @@ dir=`dirname $0`
 cd ${dir}
 
 ./tag.sh
+./identity.sh
```
Comment (jortel) [inline hack/add/all.sh:13]: Identities need to be created before being referenced by applications.

File: model/identity.go [hunk 9]
```diff
@@ -8,18 +8,16 @@ import (
...
 	Model
-	Kind          string `gorm:"not null"`
-	Name          string `gorm:"not null"`
-	Description   string
-	User          string
-	Password      string
-	Key           string
-	Settings      string
-	Encrypted     string
-	ApplicationID uint `gorm:"many2many:appIdentity"`
```
Comment (jortel) [inline model/identity.go:22]: Removed: not correct for m2m.

**Review by mansam — APPROVED** [review]
LGTM

**Pull request merged**